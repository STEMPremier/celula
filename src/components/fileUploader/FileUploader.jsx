/* eslint-disable no-unused-vars */
import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

// import { ErrorBox } from '../core';

import {
  FILEUPLOADER_TYPES as TYPES,
  COLORS,
  SIZES,
} from '../../utils/constants';

const FileUploader = ({
  // accept,
  // children,
  className,
  // capture,
  color,
  disabled,
  // errorMsg,
  formId,
  // handleClick: handler,
  helpText,
  // initialValue,
  label,
  // multiple,
  size,
  styleType,

  name,
}) => {
  // const [fieldValue, setFieldValue] = useState(initialValue);
  // const [errMsg, setErrMsg] = useState(errorMsg);

  const classes = cx(
    'ce-fileuploader',
    {
      [`ce-fileuploader--${styleType}`]: TYPES.includes(
        styleType.toString().toLowerCase(),
      ),
      [`ce-fileuploader--${color}`]: COLORS.includes(
        color.toString().toLowerCase(),
      ),
      [`ce-fileuploader--${size}`]: SIZES.includes(
        size.toString().toLowerCase(),
      ),
      'ce-fileuploader--disabled': disabled,
      // 'ce-fileuploader--error': errMsg,
    },
    className,
  );

  const id = `${name}`;

  // useEffect(() => {
  //   setFieldValue(initialValue);
  // }, [initialValue]);

  // useEffect(() => {
  //   setErrMsg(errorMsg);
  // }, [errorMsg]);

  // const handleClick = event => {
  //   const { value } = event.target;
  //   setFieldValue(value);
  //   handler(value);
  // };

  return (
    <div className={classes}>
      <label htmlFor={id}>{label}</label>
      {helpText && <div className="ce-fileuploader__help-text">{helpText}</div>}
      <input
        type="file"
        disabled={disabled}
        form={formId}
        id={id}
        name={name}
        size={size}
        color={color}
        styleType={styleType}
        // onBlur={checkValiditiy}
        // onChange={handleClick}
        // value={fieldValue}
      />
    </div>
  );
};

FileUploader.propTypes = {
  // children:
  /**
   * A class name, or string of class names, to add to the `<Input />`.
   */
  className: PropTypes.string,
  color: PropTypes.string,
  /**
   * Disables the `<Input />`.
   */
  disabled: PropTypes.bool,
  /**
   * An error message to display in the `<Input />`.
   */
  // errorMsg: PropTypes.string,
  /**
   * The id of the form the `<Input />` belongs to.
   */
  formId: PropTypes.string,
  /**
   * A function to trigger when the state of the `<Input />` changes.
   */
  // handleClick: PropTypes.func,
  /**
   * Any text to assist the user with this `<Input />`.
   */
  helpText: PropTypes.string,
  /**
   * The initial value of the `<Input />`, if any.
   */
  // initialValue: PropTypes.oneOfType([
  //   PropTypes.string,
  //   PropTypes.number,
  //   PropTypes.bool,
  // ]),
  /**
   * The `<Input />` label.
   */
  label: PropTypes.string.isRequired,
  /**
   * The name given to the `<Input />`. It connects the label to the `<Input />`.
   */
  name: PropTypes.string.isRequired,
  size: PropTypes.string,
  styleType: PropTypes.string,
};

FileUploader.defaultProps = {
  className: '',
  color: 'primary',
  disabled: false,
  // errorMsg: '',
  formId: '',
  // handleClick: () => {},
  helpText: '',
  // initialValue: null,
  size: 'large',
  styleType: 'default',
};

export default FileUploader;
