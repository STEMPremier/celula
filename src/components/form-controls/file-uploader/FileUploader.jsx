/* Copyright 2020 Tallo Inc.,
 *
 * This file is part of Celula.
 *
 * Celula is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Lesser General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */
import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

import './file-uploader.less';

const FileUploader = ({
  accept,
  className,
  capture,
  disabled,
  errorMsg,
  formId,
  handleClick: handler,
  multiple,
  name,
}) => {
  const [setFiles] = useState([]);
  const [errMsg, setErrMsg] = useState(errorMsg);

  const classes = cx(
    'ce-file-uploader',
    {
      'ce-file-uploader--disabled': disabled,
      'ce-file-uploader--error': errMsg,
    },
    className,
  );

  const id = `${name}`;

  useEffect(() => {
    setErrMsg(errorMsg);
  }, [errorMsg]);

  const handleClick = event => {
    const fs = event.target.files;

    setFiles(Object.values(fs));
    handler(event);
  };

  // const renderFiles = () => files.map(f => <span key={f.name}>{f.name}</span>);

  return (
    <div className={classes}>
      {/* complete upload file list */}
      {/* in-progress upload file list */}
      {/* <div className="ce-fu-file-list">{renderFiles()}</div> */}
      <input
        accept={accept}
        capture={capture}
        disabled={disabled}
        form={formId}
        id={id}
        multiple={multiple}
        name={name}
        onChange={handleClick}
        type="file"
      />
      {/* dotted line box container thingy */}
      <div className="ce-file-uploader__container">
        <label htmlFor={id}>Browse</label>
        &nbsp;for a file to upload
      </div>
      {errorMsg && (
        <span className="ce-file-uploader__error-text">{errorMsg}</span>
      )}
    </div>
  );
};

FileUploader.propTypes = {
  /**
   * Defines the file types the file input should accept. Inside the string, create a comma-separated list of unique file type specificiers.
   *
   * An example would be accept=".doc,.docx,application/msword,application/vnd.openxmlformats-officedocument.wordprocessingml.document".
   */
  accept: PropTypes.string,
  /**
   * Specifies which cameria to use for capture of image or video data, if the accept attribute indicates that the inupt should be one of those types.
   *
   * A value of 'user' indicates that the user-facing camera and/or microphone should be used.
   * A value of 'enviroment' specifies that the outward-facing camera and/or microphone should be used.
   * If no value is declared, the use agent is free to decide on its own what to do.
   */
  capture: PropTypes.oneOf([undefined, 'user', 'environment']),
  /**
   * A class name, or string of class names, to add to the `<FileUploader />`.
   */
  className: PropTypes.string,
  /**
   * Disables the `<FileUploader />`.
   */
  disabled: PropTypes.bool,
  /**
   * An error message to display in the `<FileUploader />`.
   */
  errorMsg: PropTypes.string,
  /**
   * The id of the form the `<FileUploader />` belongs to.
   */
  formId: PropTypes.string,
  /**
   * A function to trigger when the state of the `<FileUploader />` changes.
   */
  handleClick: PropTypes.func,
  /**
   * Allow the use to select more than one file.
   */
  multiple: PropTypes.bool,
  /**
   * The name given to the `<FileUploader />`. It connects the label to the `<FileUploader />`.
   */
  name: PropTypes.string.isRequired,
};

FileUploader.defaultProps = {
  accept: '',
  capture: undefined,
  className: '',
  disabled: false,
  errorMsg: '',
  formId: '',
  handleClick: () => {},
  multiple: false,
};

export default FileUploader;
