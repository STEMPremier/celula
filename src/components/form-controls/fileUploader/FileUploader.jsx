/* eslint-disable jsx-a11y/label-has-associated-control */
/* eslint-disable no-unused-vars */
import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

import { ErrorBox } from '../core';

import './file-uploader.less';

import {
  FILEUPLOADER_TYPES as TYPES,
  COLORS,
  SIZES,
} from '../../../utils/constants';

const FileUploader = ({
  accept,
  children,
  className,
  capture,
  color,
  disabled,
  errorMsg,
  formId,
  handleClick: handler,
  helpText,
  initialValue,
  label,
  multiple,
  name,
  size,
  styleType,
}) => {
  const [fieldValue, setFieldValue] = useState('');
  const [errMsg, setErrMsg] = useState(errorMsg);

  const classes = cx(
    'ce-fileuploader',
    {
      [`ce-fileuploader--${styleType}`]: TYPES.includes(
        styleType.toString().toLowerCase(),
      ),
      [`ce-fileuploader--${color}`]: COLORS.includes(
        color.toString().toLowerCase(),
      ),
      [`ce-fileuploader--${size}`]: SIZES.includes(
        size.toString().toLowerCase(),
      ),
      'ce-fileuploader--disabled': disabled,
      'ce-fileuploader--error': errMsg,
    },
    className,
  );

  const id = `${name}`;

  useEffect(() => {
    setErrMsg(errorMsg);
  }, [errorMsg]);

  const handleClick = event => {
    const { value } = event.target;
    setFieldValue(value);
    handler(value);
  };

  return (
    <div className={classes}>
      <label htmlFor={id}>{children}</label>
      {helpText && <div className="ce-fileuploader__help-text">{helpText}</div>}
      <input
        accept={accept}
        capture={capture}
        className={styleType}
        color={color}
        disabled={disabled}
        form={formId}
        id={id}
        multiple={multiple}
        name={name}
        onChange={handleClick}
        size={size}
        // onBlur={checkValiditiy}
        type="file"
        value={fieldValue}
      />
      {errorMsg && <ErrorBox errorMsg={errorMsg} />}
    </div>
  );
};

FileUploader.propTypes = {
  /**
   * Defines the file types the file input should accept.  Inside the string, create a comma-separated list of unique file type specificiers.  An example would be accept=".doc,.docx,application/msword,application/vnd.openxmlformats-officedocument.wordprocessingml.document".
   */
  accept: PropTypes.string,
  /**
   * Specifies which cameria to use for capture of image or video data, if the accept attribute indicates that the inut should be one of those types.  A value of 'user' indicates that the user-facing camera and/or microphone should be used.  A value of 'enviroment' specifies that the outward-facing camera and/or microphone should be used.  If no value is declared, the use agent is free to decide on its own what to do.
   */
  capture: PropTypes.string,
  /**
   *
   */
  children: PropTypes.string.isRequired,
  /**
   * A class name, or string of class names, to add to the `<Input />`.
   */
  className: PropTypes.string,
  /**
   *
   */
  color: PropTypes.string,
  /**
   * Disables the `<Input />`.
   */
  disabled: PropTypes.bool,
  /**
   * An error message to display in the `<Input />`.
   */
  errorMsg: PropTypes.string,
  /**
   * The id of the form the `<Input />` belongs to.
   */
  formId: PropTypes.string,
  /**
   * A function to trigger when the state of the `<Input />` changes.
   */
  handleClick: PropTypes.func,
  /**
   * Any text to assist the user with this `<Input />`.
   */
  helpText: PropTypes.string,
  /**
   * The initial value of the `<Input />`, if any.
   */
  initialValue: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
    PropTypes.bool,
  ]),
  /**
   * The `<Input />` label.
   */
  label: PropTypes.string.isRequired,
  /**
   * Allow the use to select more than one file.
   */
  multiple: PropTypes.bool,
  /**
   * The name given to the `<Input />`. It connects the label to the `<Input />`.
   */
  name: PropTypes.string.isRequired,
  /**
   *
   */
  size: PropTypes.string,
  /**
   *
   */
  styleType: PropTypes.string,
};

FileUploader.defaultProps = {
  accept: '',
  capture: '',
  className: '',
  color: 'primary',
  disabled: false,
  errorMsg: '',
  formId: '',
  handleClick: () => {},
  helpText: '',
  initialValue: null,
  multiple: false,
  size: 'large',
  styleType: 'default',
};

export default FileUploader;
