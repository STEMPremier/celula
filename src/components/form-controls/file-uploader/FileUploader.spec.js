import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import FileUploader from '.';

describe('<FileUploader />', () => {
  describe('render()', () => {
    test('render the FileUploader component', () => {
      const wrapper = shallow(
        <FileUploader label="test label" name="test">
          Children
        </FileUploader>,
      );
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
