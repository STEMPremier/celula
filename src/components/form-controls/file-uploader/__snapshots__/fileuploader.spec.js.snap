// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FileUploader /> render() render the FileUploader component 1`] = `
<div
  className="ce-file-uploader ce-file-uploader--primary ce-file-uploader--large"
>
  <input
    accept=""
    disabled={false}
    form=""
    id="test"
    multiple={false}
    name="test"
    onChange={[Function]}
    type="file"
    value=""
  />
  <label
    className="ce-file-uploader__button"
    htmlFor="test"
  >
    <span
      className="ce-file-uploader__label"
    >
      Children
    </span>
  </label>
</div>
`;
